### PR TITLE
[INLONG-1720] Introduce the basic framework for a workflow based on flinksql

### LINKED ISSUE #1720: [Feature] Introduce the basic framework for a workflow based on flinksql
## Description

Inlong-sort is used to extract data from different source systems, then transforms the data and finally loads the data into different storage systems.

### isolation
Each sort workflow is an independent application based on flink sql. 

### rich connectors
Flink’s Table API & SQL programs can be connected to other external systems for reading and writing both batch and streaming tables.
A table source provides access to data which is stored in external systems (such as a database, key-value store, message queue, or file system).
A table sink emits a table to an external storage system. Depending on the type of source and sink, they support different formats such as CSV, Avro, Parquet, or ORC.

If you want to implement your own custom table source or sink, have a look at [the user-defined sources & sinks page](https://nightlies.apache.org/flink/flink-docs-master/docs/dev/table/sourcessinks/).

### more
We will support customized udf, dimension table lookup in the future.

### limitations
Currently, we just support basic transform framework.

### workflow demo
```sql
CREATE TABLE events (
  f_type INT,
  f_uid INT,
  ts AS localtimestamp,
  WATERMARK FOR ts AS ts
) WITH (
  'connector' = 'datagen',
  'rows-per-second'='5',
  'fields.f_type.min'='1',
  'fields.f_type.max'='5',
  'fields.f_uid.min'='1',
  'fields.f_uid.max'='1000'
);

CREATE TABLE print_table (
  type INT,
  uid INT,
  lstmt TIMESTAMP
) WITH (
  'connector' = 'print',
  'sink.parallelism' = '2'
);

INSERT INTO print_table SELECT * FROM events where f_type = 1;
```

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by leo65535**
Fixes #1720

### Motivation

Introduce the basic framework for a workflow based on flinksql.

### Modifications

Add basic flinksql executor.

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

### Demo

Try a demo, please run `org.apache.inlog.sort.sql.examples.BasicDemoExample`.

One source, insert into two sink.

image

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

**Comment by leo65535** [comment]
cc @gosonzhang @dockerzhang, add `sort-sql` module to `inlong-sort` 

image

File: inlong-sort/sort-sql/README.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+
+# Description
+
+## overview
+Inlong-sort is used to extract data from different source systems, then transforms the data and finally loads the data into different storage systems.
```
Comment (dockerzhang) [inline inlong-sort/sort-sql/README.md:5]: the overview part should add more description for `sort-sql`.

**Comment by dockerzhang** [comment]
@leo65535 the `sort-sql` how to consume data from `cache layer`?

**Comment by ifndef-SleePy** [comment]
Hi leo, thanks for contributing! Sorry for late response.

Currently Sort module is based on Flink DataStream API. And all connectors are reimplemented. The reason here, is the scenario of Sort is much more complicated than Flink SQL implementation. Some features, like schema dynamically updating,  data metric collection, multi-tenant supporting, error handling, data quality verification are not easy to support through Flink SQL.

Actually in Tencent, we do have a platform running Flink SQL and other APIs that consumes data from TubeMQ or Pulsar. However it's not a part of InLong project. It's a highly customized platform as an internal infrastructure for Sort. However it's not open-source yet, so we changed Sort to depend on Flink Standalone cluster in InLong open-source version.

Anyway, it's hard to give this proposal an appropriate position. It's more like a parallel component to Sort. Maybe we could file an issue to discuss whether we should support another consumer parallel to Sort or not? What do you think?

**Comment by leo65535** [comment]
> Hi leo, thanks for contributing! Sorry for late response.
> 
> Currently Sort module is based on Flink DataStream API. And all connectors are reimplemented. The reason here, is the scenario of Sort is much more complicated than Flink SQL implementation. Some features, like schema dynamically updating, data metric collection, multi-tenant supporting, error handling, data quality verification are not easy to support through Flink SQL.
> 
> Actually in Tencent, we do have a platform running Flink SQL and other APIs that consumes data from TubeMQ or Pulsar. However it's not a part of InLong project. It's a highly customized platform as an internal infrastructure for Sort. However it's not open-source yet, so we changed Sort to depend on Flink Standalone cluster in InLong open-source version.
> 
> Anyway, it's hard to give this proposal an appropriate position. It's more like a parallel component to Sort. Maybe we could file an issue to discuss whether we should support another consumer parallel to Sort or not? What do you think?

hi, here is the discuss mail list.
https://lists.apache.org/thread/zsn92po39fvjktkvm2j53pz4jhgshcl7

File: inlong-sort/sort-sql/README.md (hunk 1 above)
Comment (leo65535) [reply inlong-sort/sort-sql/README.md:5]: make sense, will update it.

**Comment by leo65535** [comment]
> @leo65535 the `sort-sql` how to consume data from `cache layer`?

we use different source connectors, e.g pulsar, kafka

**Pull request closed without merging**